### DISCUSSION TITLE
With duplicated data, area mark goes a little nuts, line mark is fine

### DISCUSSION THREAD

**Discussion by pdebruic** [observablehq/plot#2193] [Q&A] [answered]
When plotting a time series with multiple instances of two (or more) data points at the same time the area plot doesn't draw a coherent shape.  

This helped me find a bug in my own stuff.  So I'm fine after removing my duplicated data. 

This is an example
https://observablehq.com/d/bd4e9b56a1163065

**Comment by mbostock** [comment]
Can you make the notebook public so we can see?

If you’re stacking the data, you’ll need to make sure the x-values are aligned. Often this is done using the bin transform (or group transform) prior to stacking.

**Reply by pdebruic** [reply]
Oh sorry.  Its public now.  I was not trying to stack.  My bug for my thing was I was mistakenly trying to plot the same value on the same date, multiple times.   Based on what was drawn it made me wonder if the area plot should've behaved in a "sane" way like the line plot.

**Reply by mbostock** [reply, marked as answer]
It’s a consequence of the implicit stacking with areaY when you use the **y** channel. If you use the **y2** channel instead (to set the topline of the area) then you can opt-out of the implicit stacking.

![untitled - 2024-10-09T184027 405](https://github.com/user-attachments/assets/f5901aa6-a5e1-48c9-8023-e385991d94e1)

```js
Plot.plot({
  marks: [
    Plot.areaY(toPlot, { x: "xVal", y2: "yVal", opacity: 0.3 }),
    Plot.line(toPlot, { x: "xVal", y: "yVal", tip: true })
  ]
})
```

**Comment by pdebruic** [comment]
Oh.  Thanks!   I didn't understand what was going on.  Thats clear.
